spec/dmv_spec.rb

```diff
@@ -40,4 +43,183 @@
       expect(@dmv.facilities_offering_service('Road Test')).to eq([@facility_2, @facility_3])
     end
   end
+
+  describe '#create_state_facilities' do
+    it 'can create default facilities array for Colorado' do
+      colorado_facilities = DmvDataService.new.co_dmv_office_locations()
+      @dmv.create_state_facilities("Colorado", colorado_facilities)
+
+      #Basic checks first
+      expect(@dmv.facilities).to be_a(Array)
+      expect(@dmv.facilities).to_not eq([])
+    end
+
+    it 'correctly create facilities array for Colorado based on API data' do
+      colorado_facilities = DmvDataService.new.co_dmv_office_locations()
+      @dmv.create_state_facilities("Colorado", colorado_facilities)
+
+      #NOTE: I previously checked specific key-value pairs in the dataset to match, but the datasets can change (they DID for MO)
+      #So, now I'll check more permanent features, but with less specificity
+      expect(@dmv.facilities[0]).to be_a(Facility)
+      expect(@dmv.facilities[0].name).to be_a(String)
+      expect(@dmv.facilities[1].address).to be_a(String)
+      expect(@dmv.facilities[2].phone.length).to eq(14)
+      expect(@dmv.facilities[0].services).to be_a(Array)
+      expect(@dmv.facilities[3].hours).to be_a(String)
+    end
+
+    it 'correctly create facilities array for New York based on API data' do
+      newyork_facilities = DmvDataService.new.ny_dmv_office_locations()
+      @dmv.create_state_facilities("New York", newyork_facilities)
+
+      # binding.pry
+
+      expect(@dmv.facilities[0]).to be_a(Facility)
+      expect(@dmv.facilities[0].name).to be_a(String)
+      expect(@dmv.facilities[1].address).to be_a(String)
+      expect(@dmv.facilities[2].phone.length).to eq(14)
+      expect(@dmv.facilities[0].services).to be_a(Array)
+      expect(@dmv.facilities[3].hours).to be_a(String)
+    end
+
+    it 'correctly create facilities array for Missouri based on API data' do
+      missouri_facilities = DmvDataService.new.mo_dmv_office_locations()
+      @dmv.create_state_facilities("Missouri", missouri_facilities)
+
+      # binding.pry
+      
+      #NOTE: this dataset list actually changed on me after I made tests; hence why I've moved to more 'permanent' testable aspects
```

Comment:
Great observation. We'll talk more about this when you get to Mod 3, but when we're using external data sources we can either:
1. Make somewhat flexible assertions or only assert on the permanent aspects OR
2. Use some mocked data instead of calling the DMVDataService. You could, for example, create a hash in this test that includes 3 facilities as they appear in the MO dataset, and that is called a "stub". You can then test for exact equivalence of different attributes once the Facility objects are created. Your "fake" facilities list hash would replace line 86, and it would be that stubbed data you pass into the create_state_facilities method

I'd argue your tests here are just fine, but sometimes we want to be able to make more robust assertions so it's helpful to use stubbed data and control exactly what those attributes will be